server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,256 @@
+            tarball_links.append(c)
+
+    if len(tarball_dirs)>0:
+        cnt = output_format(tarball_dirs, "Tarball Directories", cnt)
+    if len(empty_tarball_dirs)>0:
+        cnt = output_format(empty_tarball_dirs, "Empty tar ball directories", cnt)
+    if len(unpacking_tarball_dirs)>0:
+        cnt = output_format(unpacking_tarball_dirs, "Invalid unpacking directories", cnt)
+    if len(tarball_links)>0:
+        cnt = output_format(tarball_links, "Invalid tar ball links:", cnt)
+
+    return cnt
+
+def verify_controllers(hierarchy, config):
+    cnt=0
+    h = os.path.basename(hierarchy)
+    archive = config.ARCHIVE
+    users = config.USERS
+
+    if h == 'incoming':
+        kind="incoming"
+        user=""
+    elif h == 'results':
+        kind="results"
+        user=""
+    elif os.path.dirname(hierarchy) == users:
+        kind="results"
+        user=hierarchy
+    else:
+        print("${PROG}: verify_controllers bad argument, hierarchy_root=\"${hierarchy_root}\"\n")
+        return 1
+
+    controllers = glob.iglob(os.path.join(hierarchy,'*'))
+    unexpected_objs, unexpected_objects, mialist, empty = ([] for i in range(4))
+
+    for controller in controllers:
+        c = os.path.basename(controller)
+        if os.path.isdir(controller):
+            if not os.path.isdir(os.path.join(archive,c)):
+                mialist.append(c)
+            else:
+                states = glob.iglob(os.path.join(controller,'*'))
+                for s in states: 
+                    if os.path.isdir(s): 
+                        if len(os.listdir(s))==0:
+                            empty.append(os.path.basename(s))
+                    elif not os.path.isdir(s) and not os.path.islink(s):
+                        unexpected_objects.append(c)
+        else:
+            unexpected_objs.append(c)
+
+    if len(unexpected_objs)>0:
+        cnt = output_format(unexpected_objs, "Unexpected files found:", cnt)
+    if len(mialist)>0:
+        cnt = output_format(mialist, "Controllers which do not have a ${ARCHIVE} directory:", cnt)
+    if len(empty)>0:
+        cnt = output_format(empty, "Controllers which are empty:", cnt)
+    if len(unexpected_objects)>0:
+        cnt = output_format(unexpected_objects, "Controllers which have unexpected objects:", cnt)
+
+    return cnt
+
+def verify_users(config):
+    cnt = 0
+    users = config.USERS
+
+    users_dirs = glob.iglob(os.path.join(users,'*'))
+    user_dir, unexpected_objects = ([] for i in range(2))
+
+    for user in users_dirs:
+        u = os.path.basename(user)
+        if os.path.isdir(user):
+            user_dir.append(u)
+            cnt = verify_controllers(user,config)
+            if cnt>0:
+                cnt = cnt+1
+        else:
+            unexpected_objects.append(u)
+
+    if len(unexpected_objects)>0:
+        cnt = output_format(unexpected_objects, "Unexpected files found:", cnt)
+    
+    return cnt
+
+def output_format(n, msg, cnt):
```

Comment:
There is no need for `cnt` in `output_format()`: `cnt` counts how many things went wrong and there is nothing that is wrong here. We certainly should not increment it in this rourine, we should not even get it passed as an argument or returned as a value: the routine should just `return` with no value.